git-record/src/ui.rs

```diff
@@ -0,0 +1,1041 @@
+use std::fmt::Display;
+use std::sync::mpsc::Sender;
+
+use cursive::event::Event;
+use cursive::theme::{BaseColor, Effect};
+use cursive::traits::{Nameable, Resizable};
+use cursive::utils::markup::StyledString;
+use cursive::views::{Checkbox, Dialog, HideableView, LinearLayout, ScrollView, TextView};
+use cursive::{CursiveRunnable, CursiveRunner, View};
+use tracing::error;
+
+use crate::cursive_utils::{EventDrivenCursiveApp, EventDrivenCursiveAppExt};
+use crate::tristate::{Tristate, TristateBox};
+use crate::{FileHunks, Hunk, HunkChangedLine, RecordError, RecordState};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum HunkType {
+    Before,
+    After,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FileKey {
+    file_num: usize,
+}
+
+impl Display for FileKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Self { file_num } = self;
+        write!(f, "FileKey({})", file_num)
```

Comment:
The documentation for `Display` says:

> Display is similar to Debug, but Display is for user-facing output, and so cannot be derived.

How about making this a custom function instead?  (Maybe `view_name()` or `view_id()` or something?)

Reply:
Makes sense to me.